CrowdStrike/ember-toucan-core
# Setup ember-toucan-form to yield the Textarea Field component

## 🚀 Description

This PR adds the glue between `ember-headless-form` and `ember-toucan-core` for our form solution. This PR will add `Textarea` support so that the PR scope doesn't grow substantially.

Follow up work after this PR merges:

- Add `Input` support (similar to this PR!)
- Cut a release so we can get some feedback
- Convert to using the `<template>` tag
- Checkbox support
- CheckboxGroup support
- RadioGroup support

Usage is:

```hbs template
<ToucanForm
  class='space-y-4'
  @data={{changeset this.data this.validations}}
  @dataMode='mutable'
  @onSubmit={{this.handleSubmit}}
  @validate={{validate-changeset}}
  as |form|
>
  <form.Textarea @label='Comment' @name='comment' />

  <Button type='submit'>Submit</Button>
</ToucanForm>
```

```js component
import Component from '@glimmer/component';
import { action } from '@ember/object';
import {
  validatePresence,
  validateFormat,
} from 'ember-changeset-validations/validators';

export default class extends Component {
  data = {};

  validations = {
    comment: validatePresence(true),
  };

  handleSubmit(data) {
    console.log({ data });

    const { comment } = data;

    alert(`Form submitted with: ${comment}`);
  }
}
```

---

## 🔬 How to Test

- Green build
- View the installation docs at https://37bd9563.ember-toucan-core.pages.dev/docs/toucan-form
- Go to https://37bd9563.ember-toucan-core.pages.dev/docs/toucan-form/changeset-validation
  - Without typing anything into the field, click Submit
  - Verify errors are displayed
  - Type into the field
  - Click Submit
  - Verify errors go away and an alert is shown
- Do the same as above for https://37bd9563.ember-toucan-core.pages.dev/docs/toucan-form/native-validation

---

## 📸 Images/Videos of Functionality


https://user-images.githubusercontent.com/8069555/231857687-06e69927-5b88-4663-9fb2-f5ab6ef63e98.mov
